Sadhbh a/gcp guide - WIP (#25708)
DataDog/documentation: content/en/getting_started/integrations/google_cloud.md

      text: 'Cloud Metric Delay'
    - link: 'https://docs.datadoghq.com/agent/guide/why-should-i-install-the-agent-on-my-cloud-instances/'
      tag: 'Guide'
      text: 'Why should I install the Datadog Agent on my cloud instances?'
    - link: 'https://www.datadoghq.com/blog/gke-dashboards-integration-improvements/'
      tag: 'Blog'
      text: 'New GKE dashboards and metrics provide deeper visibility into your environment'
    - link: 'https://www.datadoghq.com/blog/google-cloud-private-service-connect/'
      tag: 'Blog'
      text: 'Access Datadog privately and monitor your Google Cloud Private Service Connect usage'
    - link: 'https://www.datadoghq.com/blog/track-bigquery-costs-performance/'
      tag: 'Blog'
      text: 'Monitor BigQuery with Datadog'
    - link: 'https://www.datadoghq.com/blog/google-cloud-cost-management/'
      tag: 'Blog'
      text: 'Empower engineers to take ownership of Google Cloud costs with Datadog'
    - link: 'https://www.datadoghq.com/blog/collect-traces-logs-from-cloud-run-with-datadog/'
      text: 'Collect traces, logs, and custom metrics from your Google Cloud Run services with Datadog'
      tag: 'Blog'
---

## Overview

> Use this guide to maximize the value of the Google Cloud integration by setting it up at the Organization or Folder level with optimized configurations and permissions.

Reviewer: let's start with the why - so customers can get started monitoring their Google Cloud environment quickly.

> Use this guide to maximize the value of the Google Cloud integration by setting it up at the Organization or Folder level with optimized configurations and permissions.

Reviewer: consider rephrasing the "maximize the value" to something simple like "monitor your".

## Setup

### Prerequisites
1) Create a [Datadog account][1]
2) Set up a [Service Account][2] in any of your Google Cloud projects
3) Review these Google Cloud Prerequisites:

{{% site-region region="us,us3,us5,eu,ap1" %}}
◆ If your organization restricts identities by domain, you must add Datadog's customer identity `C0147pk0i` as an allowed value in your policy.
{{% /site-region %}} 
◆ The Google Cloud integration requires the below APIs to be enabled **for each of the projects** you want to monitor:

> 

Reviewer: I see we are bolding "each of the projects" but I believe this can be done at the folder level? (or maybe org level too?)

Reviewer 2: Just double-checked this, and it looks like APIs have to be enabled at the project level ([reference](https://www.googlecloudcommunity.com/gc/Cloud-Hub/Enable-APIs-for-service-account-access-only-parent-project-or/m-p/607004#:~:text=You%20need%20to%20enable%20these,to%20have%20the%20required%20APIs.)).

[Cloud Monitoring API][3] 
: Allows Datadog to query your Google Cloud metric data.

[Compute Engine API][4] 
: Allows Datadog to discover compute instance data.

[Cloud Asset API][5]
: Allows Datadog to request Google Cloud resources and link relevant labels to metrics as tags.

[Cloud Resource Manager API][6] 
: Allows Datadog to append metrics with the correct resources and tags.

[IAM API][7]
: Allows Datadog to authenticate with Google Cloud.

[Google Cloud Billing API][8] 
: Enables accurate billing for API calls.

> 

Reviewer: I don't believe this is accurate based on what I see in the API. Why do we actually require it?

Reviewer: "Allows developers to manage billing for their Google Cloud Platform projects programmatically." - Are we requesting this for CCM?

> <div class="alert alert-info">You can confirm if these APIs are enabled by heading to <a href="https://console.cloud.google.com/apis/dashboard"><b>Enabled APIs & Services</a></b> under each project in your Google Cloud workspace.</div>

Reviewer: There's the "under each project" again that I think we should remove. Consider the user's predicament when they have 100s of projects.

### Organization-level metric collection

> For the most comprehensive level of monitoring in your Google Cloud environment, integrate Datadog with [Google Cloud][9] at the organization or folder level.

Reviewer: Let's start with the why here again to reiterate, and give the customer an escape hatch to monitor specific projects (i.e link to setup page setup instructions).

Org-level monitoring is recommended for comprehensive coverage of all projects including future newly created projects. The same can be achieved with folder-level monitoring.

**Note**: You must have the `Admin` role assigned to your Cloud Identity user account at the desired scope (for example, `Organization Admin`).

{{% collapse-content title="1. Create a Google Cloud service account in the default project" level="h5" %}}
1. Open your [Google Cloud console][10].
2. Navigate to **IAM & Admin** > **Service Accounts**.
3. Click **Create service account** at the top.
4. Give the service account a unique name.
5. Click **Done** to complete creating the service account.

> {{% /collapse-content %}} 

Reviewer: I get an error "Page not viewable for organizations. To view this page, select a project." (is this because I lack the permissions?)

{{% collapse-content title="2. Add the service account at the organization or folder level" level="h5" %}}
1. In the Google Cloud console, go to the **IAM** page.
2. Select a folder or organization.
3. To grant a role to a principal that does not already have other roles on the resource, click **Grant Access**, then enter the email of the service account you created earlier.
4. Enter the service account's email address.

> 5. Assign the following roles:

Reviewer: What about "Browser"?

- [Compute Viewer][52] provides **read-only** access to get and list Compute Engine resources
- [Monitoring Viewer][53] provides **read-only** access to the monitoring data availabile in your Google Cloud environment
- [Cloud Asset Viewer][54] provides **read-only** access to cloud assets metadata
6. Click **Save**.

**Note**: The `Browser` role is only required in the default project of the service account. Other projects require only the other listed roles.

> {{% /collapse-content %}}

Reviewer: I see the note here. It's confusing there's no "Browser" listed above and we have a note reading this (it makes sense in our setup page because we default to access on a single project)

{{% collapse-content title="3. Add the Datadog principal to your service account" level="h5" %}}
1. In Datadog, navigate to **Integrations** > [**Google Cloud Platform**][11].
2. Click **Add Google Cloud Account**. 
If you have no configured projects, you are automatically redirected to this page.
3. If you have not generated a Datadog principal for your org, click the **Generate Principal** button.

> 4. Copy your Datadog principal and keep it for the next section.

Reviewer: Someone can correct me but I don't think there's ever a case where the Datadog principal is *not* generated.

{{< img src="integrations/google_cloud_platform/principal-2.png" alt="The page for adding a new Google Cloud account in Datadog's Google Cloud integration tile" style="width:70%;">}}

**Note**: Keep this window open for Section 4.

5. In the [Google Cloud console][10], under the **Service Accounts** menu, find the service account you created in Section 1.
6. Go to the **Permissions** tab and click **Grant Access**.

{{< img src="integrations/google_cloud_platform/grant-access.png" alt="Google Cloud console interface, showing the Permissions tab under Service Accounts." style="width:70%;">}}

7. Paste your Datadog principal into the **New principals** text box.
8. Assign the role of **Service Account Token Creator**.
9. Click **Save**.

**Note**: If you previously configured access using a shared Datadog principal, you can revoke the permission for that principal after you complete these steps.

> {{% /collapse-content %}}

Reviewer: How/why? Given we are focusing here on a first-time user's "getting started" (not a general setup scenario?). Maybe please make this explicit context at the top.

{{% collapse-content title="4. Complete the integration setup in Datadog" level="h5" %}}
1. In your Google Cloud console, navigate to the **Service Account** > **Details** tab. On this page, find the email associated with this Google service account. It has the format `<SA_NAME>@<PROJECT_ID>.iam.gserviceaccount.com`.
2. Copy this email.
3. Return to the integration configuration tile in Datadog (where you copied your Datadog principal in the previous section).
4. In the box under **Add Service Account Email**, paste the email you previously copied.
5. Click **Verify and Save Account**.
{{% /collapse-content %}}

> After finishing these steps, metrics appear in Datadog after approximately **15 minutes**.

Reviewer: One idea is to consider pitching Google Cloud Integrations available to the customer here so they can browse it while the metrics collection will be done in the background.

> To view your metrics, use the left menu to navigate to **Metrics** > **Summary** and search for `gcp`:

Reviewer: Consider a header here. Is this about validation or monitoring key services?

{{< img src="integrations/google_cloud_platform/gcp_metric_summary.png" alt="The Metric Summary page in Datadog filtered to metrics beginning with GCP" style="width:100%;" >}}

The Google Cloud integration collects all available [Google Cloud metrics][12] from your projects through the Google Cloud Monitoring API.

{{% collapse-content title="See a sample of the Google Cloud metrics Datadog collects" level="h5" %}}

> {{% google-cloud-integrations %}}

Reviewer: These are not samples as much as they are the integrations for Google Cloud services we support. There's a list of metrics on the page we collect that we try and keep up to date.

{{% /collapse-content %}} 

### Limit metric collection filters

> To collect only the metrics that are valuable to your organization, enter tags in the **Limit Metric Collection Filters** text boxes under the **Metric Collection** tab of the Google Cloud integration tile. Only Google Cloud hosts with labels that match these defined tags are imported into Datadog. 

Reviewer: The premise is strange here. The filter by tags here are specific to filtering metrics for specific resources using tags. 

The *purpose* of these can largely be summarized to limit monitoring to specific set of hosts and/or Cloud Run instances (today).

Reviewer: @bryce an important side-note: I see we have new "limit metric collection by namespace" in the setup page. I recommend we copy that over but  please refrain from using the word "namespace" - it's made up by Datadog.

If reasonable, please also point to this release-note for the same. https://app.datadoghq.com/release-notes/optimize-metric-costs-with-granular-metric-filtering 

The purpose of this is to limit API costs for the customer by allowing granular metric collections for Google services.

In the below example, only Google Cloud hosts with the label `datadog:true` are monitored by Datadog: 

{{< img src="integrations/google_cloud_platform/limit_metric_collection.png" alt="The fields to limit metric collection in the Google Cloud integration tile" style="width:100%;" >}}

## Log collection

> Use the [Pub/Sub to Datadog template][14] to batch and compresses your log events before forwarding them to Datadog through [Google Cloud Dataflow][15]. This is the most network-efficient way to forward your logs.

Reviewer: Please call this Datadog Dataflow template. 

Link to the Cloud Architecture Center content also from here.

> Use the [Pub/Sub to Datadog template][14] to batch and compresses your log events before forwarding them to Datadog through [Google Cloud Dataflow][15]. This is the most network-efficient way to forward your logs.

Reviewer: An opening sentence on "why" collecting logs is important and also a way to specify what type of logs (i.e non-Agent logs) should be collected this way is important for the customer's understanding.

Follow [the instructions listed here][16] to set up Log Collection. For a deep dive into the benefits of the Pub/Sub to Datadog template, read [Stream your Google Cloud logs to Datadog with Dataflow][17] in the Datadog blog.

> <div class="alert alert-warning"><b>Note</b>: The <b>Dataflow API</b> must be enabled to use Google Cloud Dataflow. See <a href="https://cloud.google.com/apis/docs/getting-started#enabling_apis"><b>Enabling APIs</b></a> in the Google Cloud documentation for more information.</div>

Reviewer: nit: please stick to one style for all "Notes"

## Resource changes collection

Resource changes collection allows you to monitor changes in your Google Cloud environment. You receive resource events in Datadog when Google's Cloud Asset Inventory detects changes in your cloud resources. These events are forwarded to Datadog through a Cloud Pub/Sub topic and subscription.

For detailed setup instructions, see the [resource changes collection section][18] of the Google Cloud integration documentation.

> **Note**: The Google Cloud service account used by the integration needs the `pubsub.subscriptions.consume` permission for the Cloud Pub/Sub topic and subscription created for this purpose.

Reviewer: why are we including the notes here when we are hyperlinking to the setup instructions in the other page? Either embed both in collapsible boxes or leave out the notes also.

## Google Cloud integrations

You can send monitoring data to Datadog from 35+ integrations for Google Cloud services. The [Integrations page][19] provides a full listing of the available integrations. Many of these integrations are installed by default when Datadog recognizes data being ingested in from your Google Cloud account.

> See the blog links below for deep dives into monitoring many of the more popular services.

Reviewer: This is the most important section, let's add more details here to inform the customer *how* we help them get started quickly monitoring Google Cloud service. 

If we have time, a single line summary for each of the integrations would be amazing -- I can help or please work with our PMM, Trammell, to help cover this section. 

Tying an earlier comment - I'd like for this to be what the customer does when they wait 15 minutes for the  metrics to start flowing.

{{% collapse-content title="Integration blogs" level="h5" %}}
- [Cloud Armor][20]
- [BigQuery][21]
- [Cloud Run][22]
- [Cloud SQL][23]
- [Compute Engine][24]
- [Dataflow][25]
- [Eventarc][26]
- [Google Kubernetes Engine (GKE)][27]
- [Private Service Connect][28]
- [Security Command Center][29]
- [Vertex AI][30]
{{% /collapse-content %}}

## Leveraging the Datadog Agent

After the Google Cloud integration is configured, Datadog automatically starts collecting Google Cloud metrics. However, you can leverage the Datadog Agent to gather deeper insights into your infrastructure.

The [Datadog Agent][31] provides the [most granular, low-latency metrics][32] from your infrastructure, delivering real-time insights into CPU, memory, disk usage, and more for your Google Cloud hosts. 
The Agent can be installed on any host, including [GKE][33].

The Agent also supports a wide range of [integrations][34], enabling you to extend visibility into specific services and databases running on your hosts. 

> [Traces][35] collected through the Trace Agent enable comprehensive Application Performance Monitoring (APM), helping you understand end-to-end service performance. 

Reviewer: missing logs!

For the full list of benefits of installing the Agent on your cloud instances, see [Why should I install the Datadog Agent on my cloud instances?][36]

## Explore related services

{{% site-region region="us5,eu" %}}
### Private Service Connect 

Use the [Google Cloud Private Service Connect integration][201] to visualize connections, data transferred, and dropped packets through Private Service Connect. This gives you visibility into important metrics from your Private Service Connect connections, both for producers as well as consumers.
[Private Service Connect (PSC)][202] is a Google Cloud networking product that enables you to access [Google Cloud services][203], [third-party partner services][204], and company-owned applications directly from your Virtual Private Cloud (VPC).

> 

Reviewer: Please link to the blog. I am fine if we prefer to keep this for all regions and mention in the note this is available only for US5/EU1 regions. 

rationale: Customers sometimes care enough to migrate data centers (esp when just getting started)

---
title: Getting Started with Google Cloud
---

## Overview

> This guide provides an overview of how to get started with the Google Cloud integration with Datadog.

Reviewer 3: ```suggestion
This guide provides an overview of how to get started with the Datadog Google Cloud integration.
```

Integrating your Google Cloud projects with Datadog allows you to monitor real-time metrics & logs from your GCE and GKE compute instances.

## Prerequisites
1) Create a [Datadog account][1]
2) Set up a [Service Account][2] in Google Cloud
3) Review these Google Cloud [Prerequisites][3]:

- If your organization restricts identities by domain, you must add Datadog's customer identity `C0147pk0i` as an allowed value in your policy.
- Ensure that any projects being monitored are **not** configured as [scoping projects][42] that pull in metrics from multiple other projects.

- The Google Cloud integration requires the below APIs to be enabled **for each of the projects** you want to monitor:

[Cloud Monitoring API][7] 

> : Allows Datadog to query your Google Cloud Metric data.

Reviewer 3: ```suggestion
: Allows Datadog to query your Google Cloud metric data.
```

[Compute Engine API][8] 

> : Allows Datadog to discover Compute instance data.

Reviewer 3: ```suggestion
: Allows Datadog to discover compute instance data.
```

    - link: 'https://www.datadoghq.com/blog/google-cloud-cost-management/'
      tag: 'Blog'
      text: 'Empower engineers to take ownership of Google Cloud costs with Datadog'
    - link: 'https://www.datadoghq.com/blog/collect-traces-logs-from-cloud-run-with-datadog/'
      text: 'Collect traces, logs, and custom metrics from your Google Cloud Run services with Datadog'
      tag: 'Blog'
---

## Overview

> Use this guide to get started monitoring your Google Cloud environment at a folder or organization level. This approach simplifies the setup for Google Cloud environments with multiple projects, ensuring that there are no gaps in monitoring.

Reviewer: Remove "at a folder or organization level" at the intro. 

ensuring there are no gaps in monitoring -> instead spin this guide as a way to as maximize monitoring coverage.

: Enables accurate billing for API calls.

<div class="alert alert-info">You can confirm if these APIs are enabled by heading to <a href="https://console.cloud.google.com/apis/dashboard"><b>Enabled APIs & Services</a></b> under each project in your Google Cloud workspace.</div>

## Setup 

You can integrate Datadog with [Google Cloud][4] using [service account impersonation][5] and project discovery. Service account impersonation enables Datadog to automatically discover and monitor new projects as they are created within the assigned scope, providing seamless monitoring as your Google Cloud environment grows.

> After reviewing the above [Prerequisites][52], follow these step-by-step instructions to set up the Google Cloud integration:

Reviewer 3: Link to Prerequisites gives a 404.

{{% collapse-content title=" 1. Create your Google Cloud service account" level="h5" %}}
1. Open your [Google Cloud console][53].

2. Navigate to **IAM & Admin** > **Service Accounts**.

> 3. Click on Create service account at the top.

Reviewer 3: ```suggestion
3. Click on **Create service account** at the top.
```

: Allows Datadog to discover compute instance data.

[Cloud Asset API][5]
: Allows Datadog to request Google Cloud resources and link relevant labels to metrics as tags.

[Cloud Resource Manager API][6] 
: Allows Datadog to append metrics with the correct resources and tags.

[IAM API][7]
: Allows Datadog to authenticate with Google Cloud.

[Google Cloud Billing API][8] 
: Allows developers to manage billing for their Google Cloud Platform projects programmatically.

> 

Reviewer: Can we add an optional "Best Practices" section here to include 2 things:
1. Request "Service Usage Consumer" request here? The basic gist of it is that this allows us to attribute API calls to the appropriate project being queried. Please see https://docs.google.com/document/d/1SiAGU60P3dwCXhGN6afjSv4aOV3eOcJIqLTpJSxxDTU/edit?tab=t.0#heading=h.lfqargljeusc
Note: this requires us to allowlist the customers for this access at present (will be self-service configurable later this year)

2. Can we please ensure customers know up front that they **should not** be monitoring scoping projects with Datadog. We have this note in the main setup instructions but it needs to be emphasized for clarity. Our wording on this needs to clarify that we will not pull in metrics from metric-scopes created. We will have this change out by EOQ1 2025 (hopefully!).

{{< img src="integrations/google_cloud_platform/create-service-account.png" alt="Google Cloud console interface, showing the 'Create service account' flow. Under 'Grant this service account access to project', the four roles in the instructions are added." style="width:70%;">}}
{{% /collapse-content %}} 

{{% collapse-content title="2. Add the Datadog principal to your service account" level="h5" %}}

> 1. In Datadog, navigate to the [**Integrations** > **Google Cloud Platform**][54].

Reviewer 3: ```suggestion
1. In Datadog, navigate to **Integrations** > [**Google Cloud Platform**][54].
```

[Compute Engine][24]
: Compute Engine is a computing and hosting service that provides you with the ability to create and run virtual machines in Google Cloud.

[Dataflow][25]
: Dataflow is a fully-managed streaming analytics service that uses autoscaling and real-time data processing.

[Eventarc][26]
: Eventarc is a fully-managed service enabling you to build event-driven architectures.

[Google Kubernetes Engine (GKE)][27]
: GKE is a fully-managed Kubernetes service.

[Private Service Connect][28]
: Private Service Connect lets you access managed Google services privately from within your VPC network.

[Security Command Center][29]
: Security Command Center provides posture management and threat detection for code, identities, and data.

[Vertex AI][30]
: Vertex AI is a fully-managed generative AI development platform.
{{% /collapse-content %}}

### Limit metric collection filters

You can limit metric collection to only the specific hosts, Cloud Run instances, or Google Cloud integrations valuable to your organization. This can help control costs by reducing the number of API calls made on your behalf.

> {{% collapse-content title="Limit metric collection by Google Cloud integration" level="h4" %}}

Reviewer: I would recommend not calling this "integration" here and instead "services"? 

Can we use the wording here: https://app.datadoghq.com/release-notes/optimize-metric-costs-with-granular-metric-filtering

> | [Container Engine][23]              | Kubernetes, managed by google                                                         |

Reviewer 3: ```suggestion
| [Container Engine][23]              | Kubernetes, managed by Google                                                         |
```

> To view your metrics, navigate to `Metrics > Summary` and search for `gcp`:

Reviewer 3: ```suggestion
To view your metrics, use the left menu to navigate to **Metrics** > **Summary** and search for `gcp`:
```
I recommend removing the image showing the navigation to the metrics summary. The navigation changes all the time, so screenshots of it quickly get out of date. The text description should be enough for customers to find the page.

> You can limit the GCE/GKE instances & Cloud Run Revisions that are pulled into Datadog by entering tags in the **Limit Metric Collection Filters** text boxes under the **General** tab.

Reviewer 3: ```suggestion
You can limit the GCE/GKE instances and Cloud Run revisions that are pulled into Datadog by entering tags in the **Limit Metric Collection Filters** text boxes under the **General** tab.
```

Follow [the instructions listed here][16] to set up Log Collection. You can also use the [Stream logs from Google Cloud to Datadog][9] guide in the Google Cloud architecture center, for a more detailed explanation of the steps and architecture involved in log forwarding. For a deep dive into the benefits of the Pub/Sub to Datadog template, read [Stream your Google Cloud logs to Datadog with Dataflow][17] in the Datadog blog.

<div class="alert alert-warning">The <b>Dataflow API</b> must be enabled to use Google Cloud Dataflow. See <a href="https://cloud.google.com/apis/docs/getting-started#enabling_apis"><b>Enabling APIs</b></a> in the Google Cloud documentation for more information.</div>

## Resource changes collection

Resource changes collection allows you to monitor changes in your Google Cloud environment. You receive resource events in Datadog when Google's Cloud Asset Inventory detects changes in your cloud resources. These events are forwarded to Datadog through a Cloud Pub/Sub topic and subscription.

> For detailed setup instructions, see the [resource changes collection section][18] of the Google Cloud integration documentation.

Reviewer: Let's downgrade this to related service.

> For detailed setup instructions, see the [resource changes collection section][18] of the Google Cloud integration documentation.

Reviewer: The why is missing in this. We should update the copy to reflect the reason for why we are recommending this - "Recent Changes" in the Resource Catalog (useful for troubleshooting)

https://www.datadoghq.com/blog/recent-changes-tab/

> Our [Integrations page][26] provides a full listing of the available sub-integrations for Google Cloud. 

Reviewer 3: ```suggestion
The [Integrations page][26] provides a full listing of the available sub-integrations for Google Cloud. 
```

> ## Leveraging the Datadog Agent for greater visibility

Reviewer 3: ```suggestion
## Leveraging the Datadog Agent
```

> The [Datadog Agent][25] provides the [most granular, low-latency metrics from your infrastructure][47], delivering real-time insights into CPU, memory, disk usage, and more for your Google Cloud hosts. 

Reviewer 3: ```suggestion
The [Datadog Agent][25] provides the [most granular, low-latency metrics][47] from your infrastructure, delivering real-time insights into CPU, memory, disk usage, and more for your Google Cloud hosts. 
```

> The Agent can be installed on any host, including GKE - [Google Kubernetes Engine, Agent][28].

Reviewer 3: ```suggestion
The Agent can be installed on any host, including [GKE][28].
```

> The Agent also supports [a wide range of integrations][29], enabling you to extend visibility into specific services and databases running on your hosts. 

Reviewer 3: ```suggestion
The Agent also supports a wide range of [integrations][29], enabling you to extend visibility into specific services and databases running on your hosts. 
```

> [Traces collected through the Trace Agent][31] enable comprehensive Application Performance Monitoring (APM), helping you understand end-to-end service performance. 

Reviewer 3: ```suggestion
[Traces][31] collected through the Trace Agent enable comprehensive Application Performance Monitoring (APM), helping you understand end-to-end service performance. 
```

> For a full list of benefits with installing the Agent on your cloud instances, see [Why should I install the Datadog Agent on my cloud instances?][27]

Reviewer 3: ```suggestion
For the full list of benefits of installing the Agent on your cloud instances, see [Why should I install the Datadog Agent on my cloud instances?][27]
```

## Explore related services

### Private Service Connect 
**(Only available in US5 & US1)**

[Private Service Connect (PSC)][45] is a Google Cloud networking product that enables you to access [Google Cloud services][43], [third-party partner services][44], and company-owned applications directly from your Virtual Private Cloud (VPC).

Get started with the [Google Cloud Private Service Connect guide.][46]

### Google Cloud Run

To begin monitoring serverless functions running on Google Cloud Platform, enable the [Google Cloud Platform][6] integration. 

The [Google Cloud Run][34] page provides a list of out-of-the-box metrics collected by the Google Cloud integration, along with instructions for [Log Collection through Dataflow][35].

### Cloud Cost Management (CCM)

[Cloud Cost Management][50] provides insights for engineering and finance teams to understand how infrastructure changes impact costs, allocate spend across your organization, and identify potential improvements.

To use Google Cloud Cost Management in Datadog, [follow these steps][51].

### Security

{{< tabs >}}

> {{% tab "Cloud SIEM" %}}

Reviewer 3: What's your reason for using tabs for the two security products?

Author: Tbh I just thought it was cute & it made the page a little shorter - I've switched it back now though

[36]: https://docs.datadoghq.com/getting_started/cloud_siem/
[37]: https://console.cloud.google.com/projectselector2/security/command-center/overview?supportedpurview=organizationId,folder,project
[38]: https://docs.datadoghq.com/integrations/google_cloud_security_command_center/#installation
[39]: https://docs.datadoghq.com/security/cloud_security_management/setup/
[40]: https://docs.datadoghq.com/security/cloud_security_management/setup/cloud_integrations/?tab=googlecloud
[41]: https://docs.datadoghq.com/infrastructure/resource_catalog/
[42]: https://cloud.google.com/monitoring/settings#:~:text=A%20scoping%20project%20hosts%20a,is%20also%20a%20scoping%20project.
[43]: https://cloud.google.com/vpc/docs/private-service-connect-compatibility#google-services
[44]: https://cloud.google.com/vpc/docs/private-service-connect-compatibility#third-party-services
[45]: https://cloud.google.com/vpc/docs/private-service-connect
[46]: https://docs.datadoghq.com/integrations/google_cloud_private_service_connect/
[47]: https://docs.datadoghq.com/developers/guide/data-collection-resolution-retention/#pagetitle:~:text=n/a-,Infrastructure,-Agent%20integrations

> [48]: http://localhost:1313/getting_started/integrations/google_cloud/#explore-related-products

Reviewer 3: Fix this link.

    - link: 'https://www.datadoghq.com/blog/google-cloud-private-service-connect/'
      tag: 'Blog'
      text: 'Access Datadog privately and monitor your Google Cloud Private Service Connect usage'
    - link: 'https://www.datadoghq.com/blog/track-bigquery-costs-performance/'
      tag: 'Blog'
      text: 'Monitor BigQuery with Datadog'
    - link: 'https://www.datadoghq.com/blog/google-cloud-cost-management/'
      tag: 'Blog'
      text: 'Empower engineers to take ownership of Google Cloud costs with Datadog'
    - link: 'https://www.datadoghq.com/blog/collect-traces-logs-from-cloud-run-with-datadog/'
      text: 'Collect traces, logs, and custom metrics from your Google Cloud Run services with Datadog'
      tag: 'Blog'
---

## Overview

Integrating your Google Cloud projects with Datadog allows you to monitor your Google Cloud services within the Datadog platform.
This guide provides an overview of how to get started with the Datadog Google Cloud integration.

## Prerequisites
1) Create a [Datadog account][1]
2) Set up a [Service Account][2] in any of your Google Cloud projects
3) Review these Google Cloud Prerequisites:

{{% site-region region="us,us3,us5,eu,ap1" %}}

> - If your organization restricts identities by domain, you must add Datadog's customer identity `C0147pk0i` as an allowed value in your policy.

Reviewer: nit: can we update these dashes to be bullets?

{{% /site-region %}}

> - Datadog does not support [scoping projects][42] that pull in metrics from multiple other projects.

Reviewer: This is not a "prerequisite". We should instead note it as a warning.

- The Google Cloud integration requires the below APIs to be enabled **for each of the projects** you want to monitor:

[Cloud Monitoring API][7] 
: Allows Datadog to query your Google Cloud metric data.

[Compute Engine API][8] 
: Allows Datadog to discover compute instance data.

[Cloud Asset API][9]
: Allows Datadog to request Google Cloud resources and link relevant labels to metrics as tags.

[Cloud Resource Manager API][10] 
: Allows Datadog to append metrics with the correct resources and tags.

[IAM API][11]
: Allows Datadog to authenticate with Google Cloud.

[Google Cloud Billing API][12] 
: Enables accurate billing for API calls.

<div class="alert alert-info">You can confirm if these APIs are enabled by heading to <a href="https://console.cloud.google.com/apis/dashboard"><b>Enabled APIs & Services</a></b> under each project in your Google Cloud workspace.</div>

## Setup 

For the most comprehensive level of monitoring in your Google Cloud environment, integrate Datadog with [Google Cloud][4] using [service account impersonation][5] at the organization or folder level.

> After reviewing the above [Prerequisites][3], follow these step-by-step instructions to set up the Google Cloud integration:

Reviewer: why do we need this?

Reviewer 4: Sorry do you mean this section or this sentence?

{{% collapse-content title="1. Create a Google Cloud service account in the default project" level="h5" %}}
1. Open your [Google Cloud console][53].
2. Navigate to **IAM & Admin** > **Service Accounts**.
3. Click **Create service account** at the top.
4. Give the service account a unique name.
5. Click **Done** to complete creating the service account.
{{% /collapse-content %}} 

{{% collapse-content title="2. Add the service account at the organization or folder level" level="h5" %}}
1. In the Google Cloud console, go to the **IAM** page.
2. Select a folder or organization.
3. To grant a role to a principal that does not already have other roles on the resource, click **Grant Access**, then enter the email of the service account you created earlier.
4. Enter the service account's email address.
5. Assign the following roles:
- Compute Viewer
- Monitoring Viewer
- Cloud Asset Viewer

> - Browser

Reviewer: Yes true but why?

Reviewer 4: Are you suggesting adding a justification for each role assignment for the customer, would that be better served in our general docs instead of this getting started guide that's meant to be a quick onboarding?

6. Click **Save**.

**Note**: The `Browser` role is only required in the default project of the service account. Other projects require only the other listed roles.

> {{% /collapse-content %}}

Reviewer: What is "default" project here? (Would a customer know?)

Reviewer 4: we can probably do something like the following to make it more apparent.
```suggestion
**Note**: The `Browser` role is only required in the default project ($service-account-name@$default-project.iam.gserviceaccount.com) of the service account. Other projects require only the other listed roles.
{{% /collapse-content %}}
```

      text: 'Cloud Metric Delay'
    - link: 'https://docs.datadoghq.com/agent/guide/why-should-i-install-the-agent-on-my-cloud-instances/'
      tag: 'Guide'
      text: 'Why should I install the Datadog Agent on my cloud instances?'
    - link: 'https://www.datadoghq.com/blog/gke-dashboards-integration-improvements/'
      tag: 'Blog'
      text: 'New GKE dashboards and metrics provide deeper visibility into your environment'
    - link: 'https://www.datadoghq.com/blog/google-cloud-private-service-connect/'
      tag: 'Blog'
      text: 'Access Datadog privately and monitor your Google Cloud Private Service Connect usage'
    - link: 'https://www.datadoghq.com/blog/track-bigquery-costs-performance/'
      tag: 'Blog'
      text: 'Monitor BigQuery with Datadog'
    - link: 'https://www.datadoghq.com/blog/google-cloud-cost-management/'
      tag: 'Blog'
      text: 'Empower engineers to take ownership of Google Cloud costs with Datadog'
    - link: 'https://www.datadoghq.com/blog/collect-traces-logs-from-cloud-run-with-datadog/'
      text: 'Collect traces, logs, and custom metrics from your Google Cloud Run services with Datadog'
      tag: 'Blog'
---

## Overview

> Integrating your Google Cloud projects with Datadog allows you to monitor metrics from your GCE and GKE compute instances within the Datadog platform.

Reviewer 4: ```suggestion
Integrating your Google Cloud projects with Datadog allows you to monitor your GCE and GKE compute instances within the Datadog platform.
```

Suggesting to remove metrics from this sentence since it's the overall monitoring story which includes metrics, logs, resource changes, etc.

> This guide provides an overview of how to quickly get started with the Datadog Google Cloud integration.

Reviewer 4: ```suggestion
This guide provides an overview of how to get started with the Datadog Google Cloud integration.
```

## Prerequisites
1) Create a [Datadog account][1]

> 2) Set up a [Service Account][2] in Google Cloud

Reviewer 4: ```suggestion
2) Set up a [Service Account][2] in desired Google Cloud project
```
This is more of nit, fine to not include but want to be specific that it's in their Google Cloud environment for a project they would wantt o monitor.

3) Review these Google Cloud Prerequisites:

> - If your organization restricts identities by domain, you must add Datadog's customer identity `C0147pk0i` as an allowed value in your policy.

Reviewer 4: Can we make this identity `C0147pk0i` conditional to the site? We'll likely require a separate identity for govcloud.

- Ensure that any projects being monitored are **not** configured as [scoping projects][42] that pull in metrics from multiple other projects.

- The Google Cloud integration requires the below APIs to be enabled **for each of the projects** you want to monitor:

[Cloud Monitoring API][7] 
: Allows Datadog to query your Google Cloud metric data.

[Compute Engine API][8] 
: Allows Datadog to discover compute instance data.

[Cloud Asset API][9]

> : Allows Datadog to request Google Cloud resources and link them to relevant metric tags.

Reviewer 4: ```suggestion
: Allows Datadog to request Google Cloud resources and link relevant labels to metrics as tags.
```

[Cloud Resource Manager API][10] 
: Allows Datadog to append metrics with the correct resources and tags.

[IAM API][11]
: Allows Datadog to authenticate with Google Cloud.

[Google Cloud Billing API][12] 
: Enables accurate billing for API calls.

<div class="alert alert-info">You can confirm if these APIs are enabled by heading to <a href="https://console.cloud.google.com/apis/dashboard"><b>Enabled APIs & Services</a></b> under each project in your Google Cloud workspace.</div>

## Setup 

> Integrate Datadog with [Google Cloud][4] using [service account impersonation][5] and project discovery. Service account impersonation enables Datadog to automatically discover and monitor new projects as they are created within the assigned scope, providing seamless monitoring as your Google Cloud environment grows.

Reviewer 4: This is a little misleading service account impersonation doesn't allow for the discovery would likely rephrase like so: 
```suggestion
Integrate Datadog with [Google Cloud][4] using [service account impersonation][5] and project discovery. This enables Datadog to automatically discover and monitor new projects as they are created within the assigned scope, providing seamless monitoring as your Google Cloud environment grows.
```

    - link: 'https://www.datadoghq.com/blog/track-bigquery-costs-performance/'
      tag: 'Blog'
      text: 'Monitor BigQuery with Datadog'
    - link: 'https://www.datadoghq.com/blog/google-cloud-cost-management/'
      tag: 'Blog'
      text: 'Empower engineers to take ownership of Google Cloud costs with Datadog'
    - link: 'https://www.datadoghq.com/blog/collect-traces-logs-from-cloud-run-with-datadog/'
      text: 'Collect traces, logs, and custom metrics from your Google Cloud Run services with Datadog'
      tag: 'Blog'
---

## Overview

Integrating your Google Cloud projects with Datadog allows you to monitor your GCE and GKE compute instances within the Datadog platform.
This guide provides an overview of how to get started with the Datadog Google Cloud integration.

## Prerequisites
1) Create a [Datadog account][1]
2) Set up a [Service Account][2] in your desired Google Cloud project
3) Review these Google Cloud Prerequisites:

{{% site-region region="us,us3,us5,eu,ap1" %}}
- If your organization restricts identities by domain, you must add Datadog's customer identity `C0147pk0i` as an allowed value in your policy.
{{% /site-region %}}

> - Ensure that any projects being monitored are **not** configured as [scoping projects][42] that pull in metrics from multiple other projects.

Reviewer: I would like to say to customers we don't support scoping projects as we are rolling out this change. How do we change this framing so it doesn't sound like they (customers) have to take some action to confirm (which is the current status quo) to the future model of scoping projects simply aren't supported. (i.e that we will only pull metrics from the project itself and not any metric scopes)

cc @dtru-ddog

Author: Would something like this work? 

- Datadog does not support scoping projects that pull in metrics from multiple other projects. Only metrics from the  project itself will be ingested

- The Google Cloud integration requires the below APIs to be enabled **for each of the projects** you want to monitor:

[Cloud Monitoring API][7] 
: Allows Datadog to query your Google Cloud metric data.

[Compute Engine API][8] 
: Allows Datadog to discover compute instance data.

[Cloud Asset API][9]
: Allows Datadog to request Google Cloud resources and link relevant labels to metrics as tags.

[Cloud Resource Manager API][10] 
: Allows Datadog to append metrics with the correct resources and tags.

[IAM API][11]
: Allows Datadog to authenticate with Google Cloud.

[Google Cloud Billing API][12] 

> : Enables accurate billing for API calls.

Reviewer: @dtru-ddog what do we use this for?

Reviewer 4: We need to have a customer to enable that so that Google can bill the customer for the API calls we'll be making

<div class="alert alert-info">You can confirm if these APIs are enabled by heading to <a href="https://console.cloud.google.com/apis/dashboard"><b>Enabled APIs & Services</a></b> under each project in your Google Cloud workspace.</div>

## Setup 

> Integrate Datadog with [Google Cloud][4] using [service account impersonation][5] and project discovery. This enables Datadog to automatically discover and monitor new projects as they are created within the assigned scope, providing seamless monitoring as your Google Cloud environment grows.

Reviewer 2: Updated in [e66f3b8](https://github.com/DataDog/documentation/pull/25708/commits/e66f3b88070be76e18f8ccf6e494bae5e1e1958c), but I don't have a sandbox organization to confirm the steps. Is there one available for testing?

   {{< img src="integrations/google_cloud_platform/add-principals-blurred.png" alt="Google Cloud console interface, showing an 'Add principals' box and an 'Assign roles' interface." style="width:70%;">}}

**Note**: If you previously configured access using a shared Datadog principal, you can revoke the permission for that principal after you complete these steps.
{{% /collapse-content %}}

{{% collapse-content title="3. Complete the integration setup in Datadog" level="h5" %}}
1. In your Google Cloud console, navigate to the **Service Account** > **Details** tab. On this page, find the email associated with this Google service account. It resembles `<sa-name>@<project-id>.iam.gserviceaccount.com`.

2. Copy this email.

3. Return to the integration configuration tile in Datadog (where you copied your Datadog principal in the previous section).

4. In the box under **Add Service Account Email**, paste the email you previously copied.

5. Click on **Verify and Save Account**.
{{% /collapse-content %}}

After finishing these steps, metrics will appear in Datadog after approximately **15 minutes**.

{{% collapse-content title="(Optional) Assign roles to other projects" level="h5" %}}

> Automatic project discovery simplifies the process of adding additional projects to be monitored. If you grant your service account access to other projects, folders, or orgs, Datadog discovers these projects (and any projects nested in the folders or orgs) and automatically adds them to your integration tile.

Author: Do we recommend it?

1. Make sure you have the appropriate permissions to assign roles at the desired scope:
- Project IAM Admin (or higher)
- Folder Admin
- Organization Admin
2. In the Google Cloud console, go to the **IAM** page.
3. Select a project, folder, or organization.
4. To grant a role to a principal that does not already have other roles on the resource, click **Grant Access**, then enter the email of the service account you created earlier.
5. Assign the following roles:
- Compute Viewer
- Monitoring Viewer
- Cloud Asset Viewer (**Note**: The Browser role is only required in the default project of the service account).
6. Click **Save**.
{{% /collapse-content %}}

## Metric collection

> The Google Cloud integration collects [**all GA Google Cloud metrics**][17] from your projects through the Google Cloud Monitoring API.

Author: How does "The Google Cloud integration collects all available Google Cloud metrics from your projects through the Google Cloud Monitoring API." sound to you?

Limit the GCE/GKE instances and Cloud Run revisions that are pulled into Datadog by entering tags in the **Limit Metric Collection Filters** text boxes under the **General** tab of the Google Cloud Integration tile.
Only Google Cloud hosts with **labels** that match these defined tags are imported into Datadog. 

See Google's documentation on [Creating and managing labels][19] for more details on adding labels to your Google Cloud hosts.

In the below example, only Google Cloud hosts with the label `datadog:true` will be monitored by Datadog: 

{{< img src="integrations/google_cloud_platform/limit_metric_collection.mp4" video=true style="width:100%;" >}}

Use wildcards (`?` for single character, `*` for multi-character) to match many hosts, or `!` to exclude certain hosts. 
This example includes all c1* sized instances, but excludes staging hosts:

```text
datadog:true,env:production,!env:staging,instance-type:c1.*
```

## Log collection

Forward logs from your Google Cloud services to Datadog using [Google Cloud Dataflow][21] and the [Datadog template][22]. This method provides both compression and batching of events before they are forwarded to Datadog. 

> 

Reviewer 5: A link to the blog would be helpful @sriramkraman

> 

Reviewer 2: Updated in [e66f3b8](https://github.com/DataDog/documentation/pull/25708/commits/e66f3b88070be76e18f8ccf6e494bae5e1e1958c), thanks in advance for any additional feedback

Follow [the instructions listed here][33] to set up Log Collection.

<div class="alert alert-warning"><b>Note</b>: The <b>Dataflow API</b> must be enabled to use Google Cloud Dataflow. See <a href="https://cloud.google.com/apis/docs/getting-started#enabling_apis"><b>Enabling APIs</b></a> in the Google Cloud documentation for more information.</div>

Alternatively, to collect logs from applications running in GCE or GKE, you can also use the [Datadog Agent][32].

## Google Cloud integrations

The [Integrations page][26] provides a full listing of the available sub-integrations for Google Cloud. 
Many of these integrations are installed by default when Datadog recognizes data being ingested in from your Google Cloud account.

> 

Reviewer 2: Updated in [e66f3b8](https://github.com/DataDog/documentation/pull/25708/commits/e66f3b88070be76e18f8ccf6e494bae5e1e1958c), thanks in advance for any additional feedback

## Leveraging the Datadog Agent

After the Google Cloud integration is configured, Datadog automatically starts collecting Google Cloud metrics. However, you can leverage the Datadog Agent to gather deeper insights into your infrastructure.

The [Datadog Agent][25] provides the [most granular, low-latency metrics][47] from your infrastructure, delivering real-time insights into CPU, memory, disk usage, and more for your Google Cloud hosts. 
The Agent can be installed on any host, including [GKE][28].

The Agent also supports a wide range of [integrations][29], enabling you to extend visibility into specific services and databases running on your hosts. 

[Traces][31] collected through the Trace Agent enable comprehensive Application Performance Monitoring (APM), helping you understand end-to-end service performance. 

For the full list of benefits of installing the Agent on your cloud instances, see [Why should I install the Datadog Agent on my cloud instances?][27]

## Explore related services

### Private Service Connect 

> **(Only available in US5 & US1)**

Reviewer: it's US5 and EU1.

To use this feature, see [Getting Started with Cloud SIEM][36].

To view security findings from [Google Cloud Security Command Center][37] in Cloud SIEM, toggle the **Enable collection of security findings** option under the **Security Findings** tab & follow the setup instructions on the [Google Cloud Security Command Center guide][38].

{{< img src="integrations/google_cloud_platform/security_findings.png" style="width:90%;" >}}

### Cloud Security Management

Datadog Cloud Security Management (CSM) delivers real-time threat detection and continuous configuration audits across your entire cloud infrastructure.
Check out the [Setting up Cloud Security Management guide][39] to get started.

After setting up CSM, toggle the **Enable Resource Collection** option under the **Resource Collection** tab to start collecting configuration data for the [Resource Catalog][41] and CSM. Then, follow these instructions to enable [Misconfigurations and Identity Risks (CIEM)][40] on Google Cloud.

{{< img src="integrations/google_cloud_platform/gcp_resource_collection.png" style="width:100%;" >}}

### Database Monitoring (DBM)

> Use [Database Monitoring (DBM)][49] to gain increased insight on performance metrics, host health and query samples for your Google Cloud SQL databases.

Reviewer: do we need this?

Author: I guess not, I just thought it might be relevant for some customers since DBM is growing.

      text: 'Cloud Metric Delay'
    - link: 'https://docs.datadoghq.com/agent/guide/why-should-i-install-the-agent-on-my-cloud-instances/'
      tag: 'Guide'
      text: 'Why should I install the Datadog Agent on my cloud instances?'
    - link: 'https://www.datadoghq.com/blog/gke-dashboards-integration-improvements/'
      tag: 'Blog'
      text: 'New GKE dashboards and metrics provide deeper visibility into your environment'
    - link: 'https://www.datadoghq.com/blog/google-cloud-private-service-connect/'
      tag: 'Blog'
      text: 'Access Datadog privately and monitor your Google Cloud Private Service Connect usage'
    - link: 'https://www.datadoghq.com/blog/track-bigquery-costs-performance/'
      tag: 'Blog'
      text: 'Monitor BigQuery with Datadog'
    - link: 'https://www.datadoghq.com/blog/google-cloud-cost-management/'
      tag: 'Blog'
      text: 'Empower engineers to take ownership of Google Cloud costs with Datadog'
    - link: 'https://www.datadoghq.com/blog/collect-traces-logs-from-cloud-run-with-datadog/'
      text: 'Collect traces, logs, and custom metrics from your Google Cloud Run services with Datadog'
      tag: 'Blog'
---

## Overview

> Integrating your Google Cloud projects with Datadog allows you to monitor your GCE and GKE compute instances within the Datadog platform.

Reviewer 4: Given that most of the related integrations expand beyond GCE and GKE we should probably have that reflected in this main overview sentence
```suggestion
Integrating your Google Cloud projects with Datadog allows you to monitor your Google Cloud services within the Datadog platform.
```

This guide provides an overview of how to get started with the Datadog Google Cloud integration.

## Prerequisites
1) Create a [Datadog account][1]
2) Set up a [Service Account][2] in any of your Google Cloud projects
3) Review these Google Cloud Prerequisites:

> {{% site-region region="us,us3,us5,eu,ap1" %}}

Reviewer 4: dope

- If your organization restricts identities by domain, you must add Datadog's customer identity `C0147pk0i` as an allowed value in your policy.
{{% /site-region %}}
- Datadog does not support [scoping projects][42] that pull in metrics from multiple other projects.
- The Google Cloud integration requires the below APIs to be enabled **for each of the projects** you want to monitor:

[Cloud Monitoring API][7] 
: Allows Datadog to query your Google Cloud metric data.

[Compute Engine API][8] 
: Allows Datadog to discover compute instance data.

[Cloud Asset API][9]
: Allows Datadog to request Google Cloud resources and link relevant labels to metrics as tags.

[Cloud Resource Manager API][10] 
: Allows Datadog to append metrics with the correct resources and tags.

[IAM API][11]
: Allows Datadog to authenticate with Google Cloud.

[Google Cloud Billing API][12] 
: Enables accurate billing for API calls.

<div class="alert alert-info">You can confirm if these APIs are enabled by heading to <a href="https://console.cloud.google.com/apis/dashboard"><b>Enabled APIs & Services</a></b> under each project in your Google Cloud workspace.</div>

## Setup 

> For simplicity of setup, and to ensure there are no gaps in your monitoring, integrate Datadog with [Google Cloud][4] using [service account impersonation][5] at the organization or folder level. 

Reviewer 4: nit: sentence seems a bit wordy

Reviewer 2: Updated in [0493d54](https://github.com/DataDog/documentation/pull/25708/commits/0493d5430032e1bcf23439a28a17301b954e2c92)

4. Enter the service account's email address.
5. Assign the following roles:
- Compute Viewer
- Monitoring Viewer
- Cloud Asset Viewer
6. Click **Save**.

{{% /collapse-content %}}

{{% collapse-content title="4. Complete the integration setup in Datadog" level="h5" %}}
1. In your Google Cloud console, navigate to the **Service Account** > **Details** tab. On this page, find the email associated with this Google service account. It has the format `<SA_NAME>@<PROJECT-ID>.iam.gserviceaccount.com`.
2. Copy this email.
3. Return to the integration configuration tile in Datadog (where you copied your Datadog principal in the previous section).
4. In the box under **Add Service Account Email**, paste the email you previously copied.
5. Click on **Verify and Save Account**.

{{% /collapse-content %}}

After finishing these steps, metrics appear in Datadog after approximately **15 minutes**.

## Metric collection

The Google Cloud integration collects all available [Google Cloud metrics][17] from your projects through the Google Cloud Monitoring API.

> {{% collapse-content title="Click here for a sample of the Google Cloud metrics Datadog collects" level="h5" %}}

Reviewer 4: Is there a way to have this table be centralized since we have the same content on the [README](https://github.com/DataDog/dogweb/blob/prod/integration/google_cloud_platform/README.md?plain=1#L9) md and I can see opportunities for this to get out of sync

| [Private Service Connect][84]       | Access managed services with private VPC connections                                  |
| [Pub/Sub][85]                       | Real-time messaging service                                                           |
| [Spanner][86]                       | Horizontally scalable, globally consistent, relational database service               |
| [Storage][87]                       | Unified object storage                                                                |
| [Vertex AI][88]                     | Build, train and deploy custom machine learning (ML) models.                          |
| [VPN][89]                           | Managed network functionality                                                         |
{{% /collapse-content %}} 

You can find your Google Cloud metrics in the Metrics Summary page in the Datadog platform.
To view your metrics, use the left menu to navigate to **Metrics** > **Summary** and search for `gcp`:

{{< img src="integrations/google_cloud_platform/gcp_metric_summary.png" style="width:100%;" >}}

### Limit metric collection filters

> Limit the GCE/GKE instances and Cloud Run revisions that are pulled into Datadog by entering tags in the **Limit Metric Collection Filters** text boxes under the **General** tab of the Google Cloud Integration tile.

Reviewer 4: This has been moved to the **Metric Collection** lets update the naming accordingly.
```suggestion
Limit the GCE/GKE instances and Cloud Run revisions that are pulled into Datadog by entering tags in the **Limit Metric Collection Filters** text boxes under the **Metric Collection** tab of the Google Cloud Integration tile.
```
![Screenshot 2024-12-26 at 9 35 56 AM](https://github.com/user-attachments/assets/80272067-5787-4296-891f-a3a0e0615a7a)
